### ISSUE TITLE
Consider Running SwiftLint in CI

### ISSUE THREAD

**Issue by mattt** [mattt/Surge#83]
In the process of investigating #82, I ran `swiftlint` on the project and found 10 violations of the  `implicit_return` rule:

```
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths 
Linting 'HyperbolicTests.swift' (1/26)
Linting 'XCTestCase+Surge.swift' (2/26)
Linting 'AuxiliaryTests.swift' (3/26)
Linting 'ExponentialTests.swift' (4/26)
Linting 'MatrixTests.swift' (5/26)
Linting 'TrigonometricTests.swift' (6/26)
Linting 'ArithmeticTests.swift' (7/26)
Linting 'ConvolutionTests.swift' (8/26)
Linting 'PowerTests.swift' (9/26)
Linting 'Contents.swift' (10/26)
Linting 'Package.swift' (11/26)
Linting 'Exponential.swift' (12/26)
Linting 'ArraySliceUnsafeMemory.swift' (13/26)
Linting 'Statistics.swift' (14/26)
Linting 'Arithmetic.swift' (15/26)
Linting 'Power.swift' (16/26)
Linting 'Pointers.swift' (17/26)
Linting 'Auxiliary.swift' (18/26)
Linting 'FFT.swift' (19/26)
Linting 'UnsafeMutableMemory.swift' (20/26)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:29:9: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:30:13: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:41:9: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:42:13: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:55:9: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:56:13: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:57:17: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:69:9: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:70:13: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
/Users/mattt/Code/Surge/Sources/Surge/Pointers.swift:71:17: warning: Implicit Return Violation: Prefer implicit returns in closures. (implicit_return)
Linting 'Convolution.swift' (21/26)
Linting 'Trigonometric.swift' (22/26)
Linting 'Matrix.swift' (23/26)
Linting 'Hyperbolic.swift' (24/26)
Linting 'ArrayUnsafeMemory.swift' (25/26)
Linting 'UnsafeMemory.swift' (26/26)
Done linting! Found 10 violations, 0 serious in 26 files.
```

Currently, our CI is showing the build as passing, and project builds without warnings. Should we elevate linter diagnostics from its current advisory disposition to enforcement via a build phase or CI?

**Comment by mattt** [comment]
For whatever reason, the build phase only just kicked in for me, so I'd assumed one wasn't configured.

So I guess the real question is, should we start treating these warnings as errors?

**Comment by mattt** [comment]
Linter warnings fixed by 7b8eaa3.

**Comment by mattt** [comment]
@alejandro-isaza FWIW:

```terminal
$ swiftlint version
0.27.0
```

So yeah, everything should be fixed with my last commit.

Thanks for #84!
